channel/fd_channel.cpp

```diff
@@ -0,0 +1,147 @@
+#include "fuzzuf/channel/fd_channel.hpp"
+#include "fuzzuf/logger/logger.hpp"
+#include "fuzzuf/utils/common.hpp"
+
+#include <signal.h>
+#include <stdarg.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/wait.h>
+#include <time.h>
+#include <unistd.h>
+
+FdChannel::FdChannel() {
+    // Nothing to do
+}
+
+FdChannel::~FdChannel() {
+    TerminateForkServer();
+}
+
+int FdChannel::Send(void *buf, size_t size) {
+    int nbytes = write(forksrv_write_fd, buf, size);
+    if (nbytes < 0) {
+        ERROR("[FdChannel] Failed to send");
+    }
+    return nbytes;
+}
+
+// Recieved data to be stored to user allocated pointer `buf`
+int FdChannel::Recv(void *buf, size_t size) {
+    int nbytes = read(forksrv_read_fd, buf, size);
+    if (nbytes < 0) {
+        ERROR("[!] [FdChannel] Failed to recv");
+    }
+    return nbytes;
+}
+
+// PUT API
+// 現在PUTが実行を終了するまで待つブロッキングな実装になっています
+// Persistent mode だとプロセスが止まってfuzzerからの指示を待つらしいのでこれは実装不備
+ExecutePUTAPIResponse FdChannel::ExecutePUT() {
+    Send((void *) "ExecutePUT", 10);
+
+    ExecutePUTAPIResponse response;
+    Recv((void *) &response, sizeof(response));
+
+    DEBUG("Response { error=%d, exit_code=%d, signal_number=%d }", 
+        response.error, response.exit_code, response.signal_number);
+    // TODO: fork_server_stdout_fd, fork_server_stderr_fd からのデータ読み取り（フェーズ3預かり）
+    assert(response.error == ExecutePUTError::None);
+
+    return response;
+}
+
+// PUT API
+pid_t FdChannel::SetupForkServer(char *const pargv[]) {
```

Comment:
Should this function return `pid_t` ? I think this is a bit difficult question worth consideration.
In other words, whether the process of the created fork server should be controlled from the outside of FdChannel or the process can be touched only by FdChannel.

Reply:
This is a reasonable proporsal. 
I was doing this for extensibility from outside the module.
But, I think this is NOT required in the immediate future. So I removed this return value to avoid wrong use of FdChannel.